R/github-pat.R

```diff
@@ -44,19 +69,29 @@ set_github_pat <- function(force_new = FALSE, validate = interactive(), verbose
       }
       if(isTRUE(validate)) {
         hx <- curl::handle_setheaders(curl::new_handle(), Authorization = paste("token", cred$password))
-        req <- curl::curl_fetch_memory("https://api.github.com/user", handle = hx)
+        host <- parse_url(url)["host"]
+        api_url <- switch(
+          host,
+          `github.com` = "https://api.github.com/user",
+          sprintf("https://%s/api/v3/user", host)
+        )
+        req <- curl::curl_fetch_memory(api_url, handle = hx)
         if(req$status_code >= 400){
+          # TODO: ideally we would reveal more about the error; we have the info
```

Comment:
It would be nice to reveal more details here. We could mine this:

https://github.com/r-lib/gh/blob/85d5492f901086b46018919b8d69e53b6f1dc91c/R/gh_response.R#L42

Reply:
I think this comment re: richer error still holds.